### PR TITLE
Remove unnecessary types

### PR DESCRIPTION
**Pull request by asashour**
## Pre-launch Checklist

- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [X] I read the [Contributors Guide].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Contributors Guide]: https://github.com/flutter/samples/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

**Comment by Monassar96** [comment]
Thankyoi

File: navigation_and_routing/lib/src/data/author.dart [hunk 1]
```diff
@@ -7,7 +7,7 @@ import 'book.dart';
 class Author {
   final int id;
   final String name;
-  final List<Book> books = <Book>[];
+  final List<Book> books = [];
```
Comment (RedBrogdon) [inline navigation_and_routing/lib/src/data/author.dart:10]: The pattern we've used elsewhere is:

`final books = <Book>[];`

It's slightly more concise.

**Review by RedBrogdon — CHANGES_REQUESTED** [review]
Thanks for the PR, @asashour. I just have one change to request.

File: navigation_and_routing/lib/src/data/author.dart (hunk 1 above)
Comment (asashour) [reply navigation_and_routing/lib/src/data/author.dart]: Thanks. Yes, it is now better.

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
@RedBrogdon I believe your requested change has been implemented, but GitHub is still blocking landing the PR. Halp?

**Review by craiglabenz — COMMENTED** [review]
The Flutter framework style guide recommends always specifying types, on the idea that the code is read way more than it is written.

I think I agree with that mindset, so I'm not sure I'd call these actually "unnecessary" for our needs.

**Comment by domesticmouse** [comment]
I'm supportive of this PR, as all the types specified herin are both pretty obvious and easily inferred by the compiler.

**Comment by domesticmouse** [comment]
@RedBrogdon can you please LGTM this change to clear the change requested block?

**Review by craiglabenz — APPROVED** [review]

**Comment by craiglabenz** [comment]
You're all clear, kid. Now let's blow this thing and go home!

**Pull request merged**